### PR TITLE
Add weighted loss.

### PR DESCRIPTION
**Pull request by xyang23**

### CODE REVIEW

**Comment by louisabraham** [comment]
Thanks a lot!

Per our meeting:
- put an assertion with `isinstance` to check that the class weight is only used with a Regressor
- provide an example with imbalanced dataset, eg https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_breast_cancer.html#sklearn.datasets.load_breast_cancer
- make it work with:
    - a python list
    - a numpy array
    - a tensor that is not on the right device (test on a GPU server or make sure to call `.to(device)`)

File: lassonet/interfaces.py [hunk 1]
```diff
@@ -153,6 +155,13 @@ def __init__(
 
         self.model = None
 
+        if task == "regression" and self.weighted_loss is not None: # weighted loss is only for classifier
```
Comment (louisabraham) [inline lassonet/interfaces.py:158]: I think task is not defined. Check with `isinstance(self, LassoNetRegressor)`

**Comment by xyang23** [comment]
Just fixed this

**Comment by louisabraham** [comment]
Thanks! It looks like this PR includes other changes, can you remove them?

Also can you explain why you mean by "fix value passing"?

**Comment by xyang23** [comment]
Thanks for asking. I will remove the other changes. "Fix value passing"
means that we previously defined criterion as an abstractattr, which did
not allow us to pass values such as self.loss_weights. I added
modifications in LassoClassifier to define criterion as a property (search
keyword "property" in the code) such that we can use values from init in
the criterion function.

Best,
Xuelin

**Comment by louisabraham** [comment]
I see!
I think there is no need to do that.
criterion can be a class attribute in general and overridden as an instance
attribute if we say so in init.
You can see this as a simple way to indicate it depends on additional
parameters.

**Comment by louisabraham** [comment]
Hi, what's the status of this? Maybe it's best to close this PR and start a new one with only clean code!

**Comment by xyang23** [comment]
Thanks for checking! It would be nice to close this - I will update the
clean code some time next week.

Best,
Xuelin
